### PR TITLE
DEMO-385 use state in output data display

### PR DESCRIPTION
**Pull request by brikiernan**
**JIRA Ticket [DEMO-385](https://rocketcom.atlassian.net/browse/DEMO-385)**

## Brief Description

This pr makes use trackFile data to display it the table within the od solution tab. 

## Final Checklist

- [x] Works in Chrome
- [x] Works in Firefox
- [x] Works in Safari
- [x] Handles responsiveness as required
- [x] Dark theme styles are correct
- [x] Light theme styles are correct

### CODE REVIEW

File: src/app/mock-data/generate-data.ts [hunk 1]
```diff
@@ -244,3 +244,11 @@ export const mockTrackFiles = Object.entries(
 ).flatMap(([spacecraftId, trackFileIds], index) => {
   return trackFileIds.map((id) => generateTrackFile(spacecraftId, id, index));
 });
+
+export const randomNum = (min = 1e9, max = 9e9) => {
+  return faker.number.int({ min, max });
+};
+
+export const toPrecision = (num: number, precision = 7) => {
+  return ~~(Math.pow(10, precision) * num) / Math.pow(10, precision);
```
Comment (NicDominguez) [inline src/app/mock-data/generate-data.ts:253]: There is actually a prototype method on Number called toPrecision that does what you are trying to do without using exponents. Please use that instead.

File: src/app/main/output-data-display/output-data-display.service.ts [hunk 2]
```diff
@@ -1,64 +1,86 @@
...
+          this.deviations.push(status);
+        }
+
+        this.solutionData.push({
+          deviation: randomNum(100, 300),
+          difference: toPrecision(finalVal - value),
+          final: finalVal,
+          id: crypto.randomUUID(),
+          initial: value,
+          property: key,
```
Comment (NicDominguez) [inline src/app/main/output-data-display/output-data-display.service.ts:53]: The Orbit Properties (i.e. keys) are in camel-case and sometimes abbreviated. Please format them to be their title case full names.

File: src/app/main/output-data-display/output-data-display.service.ts [hunk 3]
```diff
@@ -1,64 +1,86 @@
...
-  constructor() {
-    // ensures most recent ods in notification
-    this.ods.sort((a, b) => b.timestamp - a.timestamp);
-    this.criticals = this.ods.filter((od) => od.status === 'critical');
-    this.diviations = this.ods.filter((od) => od.status !== 'normal');
-    this.warnings = this.ods.filter((od) => od.status === 'caution');
+  constructor(private store: Store) {
+    this.currentTrackFile$.subscribe((trackFile) => {
+      if (!trackFile) return;
+      // prettier-ignore
```
Comment (NicDominguez) [inline src/app/main/output-data-display/output-data-display.service.ts:34]: please remove the formatting ignore

**Review by NicDominguez — CHANGES_REQUESTED** [review]
Just a couple small clean up and formatting changes.

File: src/app/main/output-data-display/output-data-display.service.ts (hunk 2 above)
Comment (brikiernan) [reply src/app/main/output-data-display/output-data-display.service.ts:53]: Took care of the title case part, not sure which ones are abbreviated and what the full names would be. May need a little assistance on that part, please.

**Review by NicDominguez — APPROVED** [review]

**Pull request merged**